Replace `numpy.alltrue` with `numpy.all`

robotAstray opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary

This pull request addresses issue #75  related to the deprecation of `alltrue` in favor of `all` in numpy versions 1.25 and later. The tests in Qiskit Cold Atoms were failing when run with versions >= 1.25.

### Changes:

Replaced usages of `numpy.alltrue` with `numpy.all` throughout the codebase, to be specific in:

- `test/test_fermion_circuit_solver.py`
- `test/test_spin_circuit_solver.py`

fretchen reviewed (commented):
Cool. This looks mostly good to me. Two minor comments:

- Could you also update the requirements such that numpy is not pinned anymore ? Normally, this should be enabled by this PR.
- Could you add a reno file that describes the changes of this PR ?

robotAstray commented:
@fretchen, I have added the release notes and removed the specific version requirement for Numpy by substituting `numpy >= 1.17` with just `numpy`. (I removed the lower bound requirement so that the project works with any version of Numpy, including versions released after the current requirements was set.)

fretchen commented on requirements.txt:
```diff
@@ -1,4 +1,4 @@
-numpy>=1.17
+numpy
```
I am sorry. This was just fine. I thought that #74 was already merged. So this change is not required and the pinning that is introduced over there is getting unnecessary.

robotAstray replied:
makes sense!

fretchen reviewed (changes requested):
Revert the changes in the requirements.txt which came from a mix-up with the merge status of #74

robotAstray commented:
> Revert the changes in the requirements.txt which came from a mix-up with the merge status of #74

that's done. Thank you.

fretchen reviewed (approved):
LGTM. Thanks for the PR.
